v1/plugin/plugin.go

```diff
@@ -312,29 +507,212 @@ func startPlugin(srv server, m *meta, p *pluginProxy) int {
 	addr := fmt.Sprintf("127.0.0.1:%v", l.Addr().(*net.TCPAddr).Port)
 	lis, err := net.Listen("tcp", addr)
 	if err != nil {
-		// TODO(danielscottt): logging
-		panic(err)
+		return "", err
 	}
 	go func() {
 		err := srv.Serve(lis)
 		if err != nil {
-			panic(err)
+			log.Fatal(err)
 		}
 	}()
+	pprofAddr := "0"
+	if isPprof {
+		pprofAddr, err = startPprof()
+		if err != nil {
+			return "", err
+		}
+	}
 	resp := preamble{
 		Meta:          *m,
 		ListenAddress: addr,
 		Type:          m.Type,
-		PprofAddress:  pprofPort,
+		PprofAddress:  pprofAddr,
 		State:         0, // Hardcode success since panics on err
 	}
 	preambleJSON, err := json.Marshal(resp)
 	if err != nil {
 		panic(err)
 	}
-	libInputOutput.printOut(string(preambleJSON))
-	go p.HeartbeatWatch()
-	// TODO(danielscottt): exit code
-	<-p.halt
-	return 0
+
+	return string(preambleJSON), nil
+}
+
+// GetPluginType converts a pluginType to a string
+// describing what type of plugin this is
+func getPluginType(plType pluginType) string {
```

Comment:
It  occurs that this function was not used by mistake and should be returned back, as @kjlyon wrote:
>When -h is run, there is a section usage that has a generic message in it still: NAME: snap-plugin-collector rand - a new cli application
We should change a new cli application to be a Snap collector plugin. This is where getPluginType() was being used previously.